cea/datamanagement/database_migrator.py

```diff
@@ -0,0 +1,147 @@
+"""
+This script checks a scenario for v2.29.0 format and migrates the input tables it to the v2.31.1 format.
+
+NOTE: You'll still need to run the archetypes-mapper after this script has run.
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+import os
+import cea
+import pandas as pd
+import collections
+
+import cea.config
+import cea.inputlocator
+from cea.utilities.dbf import dbf_to_dataframe, dataframe_to_dbf
+
+__author__ = "Daren Thomas"
+__copyright__ = "Copyright 2020, Architecture and Building Systems - ETH Zurich"
+__credits__ = ["Daren Thomas"]
+__license__ = "MIT"
+__version__ = "0.1"
+__maintainer__ = "Daren Thomas"
+__email__ = "cea@arch.ethz.ch"
+__status__ = "Production"
+
+
+def find_migrators(scenario):
+    """
+    Add new migrations here as they become necessary
+    the database-migrator will run these in sequence starting from the first migrator found
+    (NOTE: I've added a dummy migration - 2.31 - 2.31.1 - to show how the principle works)
+    """
+    migrations = collections.OrderedDict()
+    migrations["v2.29.0 - v2.31.0"] = (is_2_29, migrate_2_29_to_2_31)
```

Comment:
I'm still not too sure how this works. So supposedly we change the structure again down the road, do we just add a new migration entry to the dictionary? e.g. `v2.31.0 - v3.0.0`. Then if the user has `v2.29.0` input databases, this script would run the first migration function i.e. `v2.29.0 - v2.31.0` (since it would be `true` for the first migration function) then `v2.31.0 - v3.0.0`?

I'm thinking it might not work in the long run since the shipped databases for `v2.31.0` might not be the same as the ones shipped in the new version (e.g. `v3.0.0`) so the migration function that is trying to reference older shipped databases to do its job could cause some problems (since the installed version would only contain their own version of shipped databases, unless we fetch it from somewhere maybe?).

So instead it could be `v2.29.0 - {latest}`, `v2.31.0 - {latest}`, etc, for the different database structures. Unless users _really_ want to be able to switch between different versions. But this way would also mean we have to update each migration function to see how each version would map to the latest version.

Or maybe I'm missing something

Reply:
yes exactly. the migrations would be run in sequence. but that's a really good point you mentioned. I hadn't thought of that...

I guess that means that in the migrations scripts we can't really depend on the databases being a certain way. hm...